src/button-group/internal.tsx

```diff
@@ -0,0 +1,227 @@
+      }
+      if (!containerObjectRef.current || !focusTarget) {
+        return;
+      }
+      event.preventDefault();
+
+      const focusables = getFocusablesFrom(containerObjectRef.current);
+      const activeIndex = focusables.indexOf(focusTarget);
+      handleKey(event as any, {
+        onHome: () => focusElement(focusables[0]),
+        onEnd: () => focusElement(focusables[focusables.length - 1]),
+        onInlineStart: () => focusElement(focusables[circleIndex(activeIndex - 1, [0, focusables.length - 1])]),
+        onInlineEnd: () => focusElement(focusables[circleIndex(activeIndex + 1, [0, focusables.length - 1])]),
+      });
+    }
+
+    function focusElement(element: HTMLElement) {
+      element.focus();
+    }
+
+    // List all non-disabled and registered focusables: those are eligible for keyboard navigation.
+    function getFocusablesFrom(target: HTMLElement) {
+      function isElementRegistered(element: HTMLElement) {
+        return navigationAPI.current?.isRegistered(element) ?? false;
+      }
+
+      function isElementDisabled(element: HTMLElement) {
+        if (element instanceof HTMLButtonElement) {
+          return element.disabled;
+        }
+
+        return false;
+      }
+
+      return getAllFocusables(target).filter(el => isElementRegistered(el) && !isElementDisabled(el));
+    }
+
+    return (
+      <div
+        {...baseProps}
+        className={clsx(styles.root, testUtilStyles['button-group'], baseProps.className)}
+        ref={containerRef}
+        role="toolbar"
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
+        onFocus={onFocus}
+        onBlur={onBlur}
+        onKeyDown={onKeyDown}
+      >
+        <SingleTabStopNavigationProvider
+          ref={navigationAPI}
+          navigationActive={true}
+          getNextFocusTarget={getNextFocusTarget}
+          onUnregisterFocusable={onUnregisterFocusable}
+        >
+          {items.map((itemOrGroup, index) => {
+            const content =
+              itemOrGroup.type === 'group' ? (
+                <div key={index} role="group" aria-label={itemOrGroup.text} className={styles.group}>
+                  {itemOrGroup.items.map(item => (
+                    <ItemElement
+                      key={item.id}
+                      item={item}
+                      onItemClick={onItemClick}
+                      ref={element => onSetButtonRef(item, element)}
+                    />
+                  ))}
+                </div>
+              ) : (
+                <ItemElement
+                  item={itemOrGroup}
+                  onItemClick={onItemClick}
+                  ref={element => onSetButtonRef(itemOrGroup, element)}
+                />
+              );
+
+            const isGroupBefore = items[index - 1]?.type === 'group';
+            const isGroupNow = items[index]?.type === 'group';
+            const shouldAddDivider = isGroupBefore || (!isGroupBefore && isGroupNow && index !== 0);
```

Comment:
Let's add a dev warning (with warnOnce) for empty groups

Reply:
Added